src/main/kotlin/com/openlattice/organization/OrganizationExternalDatabaseApi.kt

```diff
@@ -0,0 +1,167 @@
+
+        const val ID = "id"
+        const val ID_PATH = "/{$ID}"
+        const val TABLE_NAME = "tableName"
+        const val TABLE_NAME_PATH = "/{$TABLE_NAME}"
+        const val TABLE_ID = "tableID"
+        const val TABLE_ID_PATH = "/{$TABLE_ID}"
+        const val COLUMN_NAME = "columnName"
+        const val COLUMN_NAME_PATH = "/{$COLUMN_NAME}"
+        const val ROW_COUNT = "rowCount"
+        const val ROW_COUNT_PATH = "/{$ROW_COUNT}"
+        const val USER_ID = "userId"
+        const val USER_ID_PATH = "/{$USER_ID}"
+        const val CONNECTION_TYPE = "connectionType"
+        const val CONNECTION_TYPE_PATH = "/{$CONNECTION_TYPE}"
+    }
+
+    /**
+     *
+     * @param organizationId The organization's UUID
+     * @param userId The user to be given trusted status
+     * @param connectionType The connection type via which the user will be trusted
+     * @param ipAddresses The set of ip addresses (as strings) from which users
+     * will be trusted. This should be an empty set if the connectionType is local.
+     */
+    @POST(BASE + ID_PATH + USER_ID_PATH + CONNECTION_TYPE_PATH + EXTERNAL_DATABASE)
+    fun addTrustedUser(
+            @Path(ID) organizationId: UUID,
+            @Path(USER_ID) userId: String,
+            @Path(CONNECTION_TYPE) connectionType: String,
+            @Body ipAddresses: Set<String>)
+
+    @DELETE(BASE + ID_PATH + USER_ID_PATH + EXTERNAL_DATABASE)
+    fun removeTrustedUser(
+            @Path(ID) organizationId: UUID,
+            @Path(USER_ID) userId: String)
+
+    //will need to add an update trusted user as well
+
+    //get
+
+    /**
+     * Gets all OrganizationExternalDatabaseTable objects for an organizatoin
+     * @param organizationId The organization's UUID
+     */
+    @GET(BASE + ID_PATH + EXTERNAL_DATABASE_TABLE)
+    fun getExternalDatabaseTables(@Path(ID) organizationId: UUID): Set<OrganizationExternalDatabaseTable>
+
+    /**
+     * Gets all OrganizationExternalDatabaseTable objects mapped to
+     * their containing OrganizationExternalDatabase Columns
+     * for an organization
+     * @param organizationId The organization's UUID
+     */
+    @GET(BASE + ID_PATH + EXTERNAL_DATABASE_TABLE + EXTERNAL_DATABASE_COLUMN)
+    fun getExternalDatabaseTablesWithColumns(@Path(ID) organizationId: UUID): Map<OrganizationExternalDatabaseTable, Set<OrganizationExternalDatabaseColumn>>
+
+    /**
+     * Gets an OrganizationExternalDatabaseTable object mapped to
+     * their containing OrganizationExternalDatabase Columns
+     * for an organization
+     * @param organizationId The organization's UUID
+     * @param tableId The id of the organization's table
+     */
+    @GET(BASE + ID_PATH + TABLE_ID_PATH + EXTERNAL_DATABASE_TABLE + EXTERNAL_DATABASE_COLUMN)
+    fun getExternalDatabaseTableWithColumns(@Path(ID) organizationId: UUID, @Path(TABLE_ID) tableId: UUID): Pair<OrganizationExternalDatabaseTable, Set<OrganizationExternalDatabaseColumn>>
```

Comment:
Avoid returning pair as this will just return ```{ "first": .... , "second": ... }```. Just do a Kotlin data class with more semantic names.

Reply:
For the similar api that returns a map instead of a single pair, for consistency would it be better to return a set of this new data class?